source/common/http/http2/codec_impl.cc

```diff
@@ -166,6 +166,11 @@ ConnectionImpl::StreamImpl::StreamImpl(ConnectionImpl& parent, uint32_t buffer_l
 }
 
 void ConnectionImpl::StreamImpl::destroy() {
+  // Cancel any pending buffered data callback for the stream.
+  if (process_buffered_data_callback_ != nullptr && process_buffered_data_callback_->enabled()) {
```

Comment:
would it make sense to cancel in the data callback destructor and just reset() here?